versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/FetchValuesUsingOpenAddressing.java

```diff
@@ -165,6 +165,13 @@ Collection<Key> checkForKeys(
 
       KeyListEntry[] keys = keyListsArray[segment];
       for (int i = offsetInSegment; ; i++) {
+        if (keys.length == 0 && segment == 0) {
```

Comment:
Move out of the loop?

Reply:
The logic becomes more complicated, IMHO, if we do that check out of the loop. We have to add an `else` around the `for`... Another `break` case aligns nicely with the previous `break` cases. In principle, the condition here is about handling a "not found" case in addition to the other not found cases.

I can move it out, if you prefer, though.

Reply:
that sgmt :+1:

Reply:
updated